lib/signs/realtime.ex

```diff
@@ -133,6 +133,7 @@ defmodule Signs.Realtime do
         current_time,
         alert_status
       )
+      |> add_route_if_alert(sign_routes)
```

Comment:
That one I'd argue against. I think it's better to have all the data fetching happen at the top level, both for clarity reasons and because it makes it easier to pass the data where it's needed, so we don't end up double-fetching.

Reply:
So then would the idea your suggesting require us to pass `sign_routes` from the top level into `get_messages` to avoid double fetching? Part of the reason why I set it up the current was was to avoid changing the method signature of `get_messages` and incur a bunch of unit test and type spec changes that I felt were better avoided

Reply:
I was just thinking to pass `sign` from `get_headway_or_alert_messages` to `get_alert_messages` and call `SourceConfig.sign_routes` in there.

Just to back up and untangle a couple layers of conversation here:
* The goal of the original suggestion was to make the message computation as clear as possible, by including the routes at the time it's created rather than modifying it later on in the call chain.
* I think computing the route list twice isn't a problem, because it's a pure function on a small immutable data structure, so it should be very cheap.
* I think we need to be more careful around getting the alert status, because it's fetching from another process. That means it's probably more expensive, but also there's a temporal aspect: we're fetching at a point in time, so if we do it twice we might get different results. (Not that that would happen if we just moved the call down the call chain, but it would increase the risk of drifting into that situation in the future, like we had with fetching the predictions recently)
* Regarding the tests, I'd think it's reasonable that there would be _some_ message test changes, given that the behavior of the system is changing.

Reply:
Gotcha. Yeah that all makes sense to me! Definitely see the point about keeping the alerts status computation at the top level.